src/utils/__tests__/httpSignature/ec.test.ts

```diff
@@ -0,0 +1,210 @@
+
+const CHALLENGE_SIGNATURE_BASE = `"${SIGNATURE_HEADER_NAME}": 2a6a0a73efb1197847f2426d3b508411688ddc924248cde9aae0911aad73a676
+"@signature-params": ("${SIGNATURE_HEADER_NAME}");created=1678475063;nonce="nonce-123";alg="ecdsa-p256-sha256";keyid="bLYYuCTAYVd7hEgxjGBkoyn1u6ztoSPlJZ5Oof6r3D4"`;
+
+const TEST_CONTENT = [
+  {
+    header: TOS_HEADER_NAME,
+    signatureBase: TOS_CHALLENGE_SIGNATURE_BASE,
+    challenge: TOS_CHALLENGE
+  },
+  {
+    header: SIGNATURE_HEADER_NAME,
+    signatureBase: CHALLENGE_SIGNATURE_BASE,
+    challenge: CHALLENGE
+  }
+];
+
+describe("Test JWK utilities", () => {
+  it("Test JWK thumbprint to be right", async () => {
+    const thumbprint = await jose.calculateJwkThumbprint(
+      ecPublicKeyJwk,
+      "sha256"
+    );
+    expect(thumbprint).toBe(ecThumbprint);
+  });
+
+  it("Test JWK thumbprint to be wrong", async () => {
+    const thumbprint = await jose.calculateJwkThumbprint(
+      ecPublicKeyJwk,
+      "sha512"
+    );
+    expect(thumbprint).not.toBe(ecThumbprint);
+  });
+
+  it("Test JWK to PEM to be right", async () => {
+    const pemKey = await pipe(
+      toPem(ecPublicKeyJwk),
+      TE.fold(
+        () => T.of(""),
+        result => T.of(result)
+      )
+    )();
+    expect(pemKey).toBe(ecPublicKeyPem);
+  });
+
+  it("Test JWK to PEM to be wrong", async () => {
+    const pemKey = await pipe(
+      toPem({ ...ecPublicKeyJwk, x: "" }),
+      TE.fold(
+        () => T.of(""),
+        result => T.of(result)
+      )
+    )();
+    expect(pemKey).not.toBe(ecPublicKeyPem);
+  });
+});
+
+describe("Test FCI custom content TOS and Sign Challenges", () => {
+  TEST_CONTENT.forEach(content => {
+    it(`Test FCI custom content to sign for "${content.header}" to be computed the right way`, async () => {
+      const customContentSignatureBase = getCustomContentSignatureBase(
+        SIGNATURE_INPUT,
+        content.challenge,
+        content.header
+      );
+      expect(customContentSignatureBase!.signatureBase).toBe(
+        content.signatureBase
+      );
+    });
+  });
+
+  TEST_CONTENT.forEach(content => {
+    it(`Test FCI custom content signature for "${content.header}" to be verified the right way`, async () => {
+      const customContentSignatureBase = getCustomContentSignatureBase(
+        SIGNATURE_INPUT,
+        content.challenge,
+        content.header
+      );
+
+      const customContentChallenge = getCustomContentChallenge(
+        customContentSignatureBase!.signatureLabel!,
+        SIGNATURE
+      );
+
+      const result = await verifyCustomContentChallenge(
+        customContentSignatureBase!.signatureBase,
+        customContentChallenge!,
+        ecPublicKeyJwk
+      )();
+
+      expect(result).toBeTruthy();
+    });
+  });
+});
+
+describe("Test provided signature algorithms", () => {
+  TEST_CONTENT.forEach(content => {
+    it(`Test that we retrive a valid sign algorithm for "${content.header}" from signature-input`, () => {
```

Comment:
What do you think about write test description like: `it should retrive a valid sign algorithm for ....`
```suggestion
    it(`it should retrive a valid sign algorithm for "${content.header}" from signature-input`, () => {
```

Reply:
Yup, simply better!